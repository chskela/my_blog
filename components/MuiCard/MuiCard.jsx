import React from "react";
import cn from "classnames";
import Link from "next/link";
import Image from "next/image";
import PropTypes from "prop-types";

import classes from "./class.module.css";

import { MuiCardDescription } from "../index";

const MuiCard = React.memo(({ article, className }) => {
  const { category, title, author, date, excerpt, url, slug } = article;
  const src = `/assets/blog/${slug}/${url}`;

  return (
    <Link href={`/posts/${slug}`} passHref>
      <article className={cn(classes.article, className)}>
        <div data-role="image">
          <Image
            src={src}
            alt={title}
            width={370}
            height={300}
            layout="responsive"
          />
        </div>
        <div data-role="description">
          <MuiCardDescription
            date={date}
            title={title}
            author={author}
            excerpt={excerpt}
            category={category}
          />
        </div>
      </article>
    </Link>
  );
});

MuiCard.propTypes = {
  article: PropTypes.shape({
    category: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    author: PropTypes.shape({
      name: PropTypes.string.isRequired,
      picture: PropTypes.string.isRequired,
    }),
    date: PropTypes.string.isRequired,
    excerpt: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    slug: PropTypes.string.isRequired,
  }),
  className: PropTypes.string,
};

MuiCard.defaultProps = {
  className: "",
};

export default MuiCard;
