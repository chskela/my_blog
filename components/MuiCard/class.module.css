.article {
  display: flex;
  cursor: pointer;
  flex-direction: row;
  align-items: center;
  margin-bottom: var(--size-14);
  justify-content: space-between;
}

.article > div[data-role="image"] {
  flex: 46;
}

.article > div[data-role="description"] {
  flex: 54;
  margin-left: var(--size-14);
}
