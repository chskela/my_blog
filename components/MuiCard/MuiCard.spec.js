import React from "react";
import { shallow } from "enzyme";
import MuiCard from "./MuiCard";

const setUp = (props) => shallow(<MuiCard {...props} />);

describe("MuiCard component", () => {
  const props = {
    article: {
      category: "test_category",
      title: "test_title",
      author: {
        name: "test_name",
        picture: "test_picture",
      },
      date: "test_date",
      excerpt: "test_excerpt",
      url: "test_url",
      slug: "test_slug",
    },
  };

  it("should render MuiCard component default props", () => {
    const component = setUp(props);
    expect(component).toMatchSnapshot();
  });
  it("should render MuiCard component whis props", () => {
    const component = setUp({ ...props, className: "test" });
    expect(component).toMatchSnapshot();
  });
});
