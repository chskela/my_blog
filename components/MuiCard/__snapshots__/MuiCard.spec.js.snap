// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MuiCard component should render MuiCard component default props 1`] = `
<Link
  href="/posts/test_slug"
  passHref={true}
>
  <article
    className="article"
  >
    <div
      data-role="image"
    >
      <Image
        alt="test_title"
        height={300}
        layout="responsive"
        src="/assets/blog/test_slug/test_url"
        width={370}
      />
    </div>
    <div
      data-role="description"
    >
      <Memo()
        author={
          Object {
            "name": "test_name",
            "picture": "test_picture",
          }
        }
        category="test_category"
        date="test_date"
        excerpt="test_excerpt"
        title="test_title"
      />
    </div>
  </article>
</Link>
`;

exports[`MuiCard component should render MuiCard component whis props 1`] = `
<Link
  href="/posts/test_slug"
  passHref={true}
>
  <article
    className="article test"
  >
    <div
      data-role="image"
    >
      <Image
        alt="test_title"
        height={300}
        layout="responsive"
        src="/assets/blog/test_slug/test_url"
        width={370}
      />
    </div>
    <div
      data-role="description"
    >
      <Memo()
        author={
          Object {
            "name": "test_name",
            "picture": "test_picture",
          }
        }
        category="test_category"
        date="test_date"
        excerpt="test_excerpt"
        title="test_title"
      />
    </div>
  </article>
</Link>
`;
